### DISCUSSION TITLE
How to open another window while hiding the root window?

### DISCUSSION THREAD

**Discussion by 24ocsicnarf** [canton7/Stylet#304] [Q&A]
I have a root window in which the password will be entered. Then, the 'another' window will show up and the root window will be hidden for a while. When I close the 'another' window, the root window should show up again.

Thanks!

**Comment by kezara** [comment]
it's maybe late, and probably you've solved your issue, but I've done something similar with Navigation controller sample, try that...

**Comment by canton7** [comment]
You need some code, somewhere, which decides when to show and hide the different windows, and you need a way to tell it when e.g. the password has been entered.

This could be two ViewModels sending messages to each other through the EventAggregator (VM 1 sends "I've got a password" and hides itself; VM 2 gets this message and shows itself), or you could have a central service which they both talk to (VM 1 calls a method on the central service saying "I've got a password", and the central service tells VM 1 to hide itself and VM 2 to show itself).
